src/main/scala/org/hammerlab/guacamole/readsets/ReadSets.scala

```diff
@@ -24,49 +24,76 @@ object ReadSets {
             contigLengthsFromDictionary: Boolean = true,
             config: Read.ReadLoadingConfig = Read.ReadLoadingConfig.default): ReadSets = {
 
-    val readsAndDicts =
+    val (readRDDs, sequenceDictionaries) =
       filenames.map(filename => {
-
         Read.loadReadRDDAndSequenceDictionary(
           filename,
           sc,
           filters = filters,
           config
         )
-      })
-
-    val (readRDDs, sequenceDictionaries) = readsAndDicts.unzip
+      }).unzip
 
-    val sequenceDictionary = sequenceDictionaries.head
-    sequenceDictionaries.tail.foreach(sd =>
-      assert(
-        sd == sequenceDictionary,
-        s"Sequence dictionaries differ:\n${sequenceDictionaries.mkString("\n\n")}"
-      )
-    )
+    val sequenceDictionary = mergeSequenceDictionaries(filenames, sequenceDictionaries)
 
     val contigLengths: ContigLengths = {
       if (contigLengthsFromDictionary) {
         getContigLengthsFromSequenceDictionary(sequenceDictionary)
       } else {
         sc.union(readRDDs)
-        .flatMap(_.asMappedRead)
-        .map(read => read.referenceContig -> read.end)
-        .reduceByKey(math.max)
-        .collectAsMap()
-        .toMap
+          .flatMap(_.asMappedRead)
+          .map(read => read.referenceContig -> read.end)
+          .reduceByKey(math.max)
+          .collectAsMap()
+          .toMap
       }
     }
 
     ReadSets(readRDDs.map(ReadsRDD(_)).toVector, sequenceDictionary, contigLengths)
   }
 
+  def mergeSequenceDictionaries(filenames: Seq[String], dicts: Seq[SequenceDictionary]): SequenceDictionary = {
```

Comment:
I think a header comment here explaining what sequence dictionaries are and why we're merging them would be useful